src/main/java/blusunrize/immersiveengineering/common/blocks/metal/BottlingMachineBlockEntity.java

```diff
@@ -219,33 +178,51 @@ public void onEntityCollision(Level world, Entity entity)
 			BottlingMachineBlockEntity master = master();
 			if(master==null)
 				return;
-			ItemStack stack = itemEntity.getItem();
-			if(stack.isEmpty())
+
+			List<Tuple<ItemEntity, ItemStack>> itemsOnConveyor = level.getEntitiesOfClass(
+					ItemEntity.class, AABB.unitCubeFromLowerCorner(Vec3.atLowerCornerOf(getBlockPos()))
+			).stream().map(itemEntity1 -> new Tuple<>(itemEntity1, itemEntity1.getItem())).toList();
```

Comment:
I'd replace Tuple either by `Pair` (the one from DFU, everything else in IE uses that) or a quick `record` with proper member names.